django/db/backends/base/operations.py

```diff
@@ -269,6 +269,17 @@ def limit_offset_sql(self, low_mark, high_mark):
             if sql
         )
 
+    def fk_on_delete_sql(self, operation):
+        """
+        Return the SQL to make an ON DELETE statement during a CREATE TABLE
+        statement.
+        """
+        if operation in ["CASCADE", "SET NULL", "RESTRICT", "SET DEFAULT"]:
+            return f" ON DELETE {operation} "
+        if operation == "":
+            return ""
```

Comment:
nit:

```suggestion
        if operation == "":
            return ""
        if operation in ("CASCADE", "SET NULL", "RESTRICT", "SET DEFAULT"):
            return f" ON DELETE {operation} "
```

This would be a bit faster.

Reply:
Not really necessary... CPython’s compiler already converts lists used only for `in` into tuples:

```
In [1]: import dis

In [2]: def f():
   ...:     1 in (1, 2)
   ...:

In [3]: def g():
   ...:     1 in [1, 2]
   ...:

In [4]: dis.dis(f)
  1           0 RESUME                   0

  2           2 LOAD_CONST               1 (1)
              4 LOAD_CONST               2 ((1, 2))
              6 CONTAINS_OP              0
              8 POP_TOP
             10 LOAD_CONST               0 (None)
             12 RETURN_VALUE

In [5]: dis.dis(g)
  1           0 RESUME                   0

  2           2 LOAD_CONST               1 (1)
              4 LOAD_CONST               2 ((1, 2))
              6 CONTAINS_OP              0
              8 POP_TOP
             10 LOAD_CONST               0 (None)
             12 RETURN_VALUE

```

See how `g` also uses `LOAD_CONST` with constant 2, a tuple.

Reply:
Oh, django now support 3.10+, then - yes, `list` can stay :)

Reply:
This isn't a 3.8 thing, it goes back to Python 2.4 and improved in 3.1! https://github.com/python/cpython/commit/b7fbcd396fc4a366433cf6f26cae64fecb056099 / https://github.com/adamchainz/cpython/commit/2c31a058eb719da8908ea9d9c2f1b748ecc7a4ca

Reply:
Wow, time flies :)
I though it was 3.8 or something.